cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -351,21 +352,24 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
           twillPreparer.setSchedulerQueue(schedulerQueueName);
         }
 
-        // Set JVM options based on configuration
+        // Set JVM options based on configuration. Skip adding JVM options for tethering provisioner as they are
+        // added in AbstractRuntimeTwillPreparer's constructor
         String jvmOpts = cConf.get(Constants.AppFabric.PROGRAM_JVM_OPTS);
```

Comment:
I would have thought things would work if we just look in the options for any additional jvm options and add them to the twillPreparer here. Not clear on why we need all these other changes and why we need to special case tethering.

Reply:
> we just look in the options for any additional jvm options and add them to the twillPreparer here

I will try this and update here.

> why we need to special case tethering.

Couple of reasons to special case tethering:
1. Avoid adding jvm options that are not relevant to running on k8s - -DCDAP_LOG_DIR=, -Dlogback.configurationFile==resources.jar/resources/logback.xml, -Dtwill.container.class.loader
2. Avoid adding `app.program.jvm.opts` twice to the twillPreparer. It's first added in the twillPreparer's [constructor](https://github.com/cdapio/cdap/blob/3435c5a5d3f2e975792336a8d64aef1a85092a5e/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/AbstractRuntimeTwillPreparer.java#L145). So avoid re-adding `app.program.jvm.opts` to the twillPreparer here.

Reply:
Is it already getting added twice? If so it seems like it shouldn't be getting added in the constructor?

Reply:
for the first point, is this an existing issue? If so it would be better to do that in a separate PR rather than mix up multiple fixes in this one.

Reply:
- removed jvm options addition in constructor
- reverted the logback change